neonvm/runner/main.go

```diff
@@ -807,6 +815,86 @@ func listenForCPUChanges(ctx context.Context, logger *zap.Logger, port int32, cg
 	}
 }
 
+func drainLogsReader(reader *bufio.Reader) error {
+	for {
+		slice, err := reader.ReadSlice('\n')
+		if err != nil {
+			if len(slice) != 0 {
+				// Print and add a newline to the slice we got
+				fmt.Printf("%s\n", slice)
+			}
+
+			return err
+		}
+
+		fmt.Printf("%s", slice)
+	}
+}
+
+// writes from socket to stdout line by line
+func forwardLogs(ctx context.Context, logger *zap.Logger, wg *sync.WaitGroup) {
+	defer wg.Done()
+
+	delay := 3 * time.Second
+	var conn net.Conn
+	var reader *bufio.Reader
+	b := &backoff.Backoff{
+		Min:    100 * time.Millisecond,
+		Max:    delay,
+		Factor: 2,
+		Jitter: true,
+	}
+
+	tryForward := func() {
```

Comment:
I found the control flow here confusing. WDYT about moving `tryForward` so it's defined and immediately called inside the loop? something like
```go
func () {
    // ...
}()
```

Reply:
sure, d65efa744be2d0a9e288f2b972b7224d5d5acb81